smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/EventStreamGenerator.java

```diff
@@ -356,7 +356,7 @@ private String getEventHeaderType(Shape shape) {
             case BLOB:
                 return "binary";
             default:
-                return "binary";
+                throw new IllegalArgumentException("Unsupported event header shape type: " + shape.getType());
```

Comment:
there should be no default, since all supported types are accounted for, see https://smithy.io/2.0/spec/streaming.html#eventheader-trait